src/log-component.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ExtensionStarterBindings} from './keys';
+import {Component, ProviderMap} from '@loopback/core';
+import {LogProvider} from './providers/log-provider';
+
+export class LogComponent implements Component {
+  providers?: ProviderMap;
+
+  constructor() {
+    this.providers = {
```

Comment:
I'd move this up to the declaration above. That way, if someone wants to make `FooLogComponent` and extend this component, they're not forced to instantiate the providers map twice (they'll just override the definition themselves).

Reply:
In hindsight, I did this same thing in one of my own PRs, so I should probably open a minor PR to correct it. :P